language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/createvar/CreateVariableWithTypeCodeAction.java

```diff
@@ -0,0 +1,283 @@
+                boolean isTuple = type.startsWith("[") && type.endsWith("]") && !type.endsWith("[]");
+                String typeLabel = isTuple && type.length() > 10 ? "Tuple" : type;
+                commandTitle = String.format(CommandConstants.CREATE_VARIABLE_TITLE + " with '%s'", typeLabel);
+            }
+
+            actions.add(CodeActionUtil.createCodeAction(commandTitle, edits, uri, CodeActionKind.QuickFix));
+        }
+        return actions;
+    }
+
+    @Override
+    protected CreateVariableCodeAction.CreateVariableOut getCreateVariableTextEdits(Range range,
+                                                                                    DiagBasedPositionDetails posDetails,
+                                                                                    TypeSymbol typeDescriptor,
+                                                                                    CodeActionContext context,
+                                                                                    ImportsAcceptor importsAcceptor) {
+        Symbol matchedSymbol = posDetails.matchedSymbol();
+
+        Position position = PositionUtil.toPosition(posDetails.matchedNode().lineRange().startLine());
+        Set<String> allNameEntries = context.visibleSymbols(position).stream()
+                .filter(s -> s.getName().isPresent())
+                .map(s -> s.getName().get())
+                .collect(Collectors.toSet());
+
+        String name = NameUtil.generateVariableName(matchedSymbol, typeDescriptor, allNameEntries);
+
+        List<TextEdit> edits = new ArrayList<>();
+        List<Integer> renamePositions = new ArrayList<>();
+        List<String> types = getPossibleTypes(typeDescriptor, context);
+        Position pos = range.getStart();
+        for (String type : types) {
+            Position insertPos = new Position(pos.getLine(), pos.getCharacter());
+            String edit = type + " " + name + " = ";
+            edits.add(new TextEdit(new Range(insertPos, insertPos), edit));
+        }
+        return new CreateVariableCodeAction.CreateVariableOut(name, types, edits,
+                importsAcceptor.getNewImportTextEdits(), renamePositions);
+    }
+
+    /**
+     * Get all possible return type combinations for the type infer required method.
+     * Here if the type symbol is a union we will check it contains an error member.
+     * Other members of the union will then combine with the error type and produce the output.
+     * In here we will `any` type.
+     *
+     * @param typeSymbol Return type descriptor of the method.
+     * @param context    CodeActionContext
+     * @return {@link List<String>}
+     */
+    private List<String> getPossibleTypes(TypeSymbol typeSymbol, CodeActionContext context) {
+        typeSymbol = getRawType(typeSymbol, context);
+        Set<String> possibleTypes = new HashSet<>();
+        List<String> finalPossibleTypeSet = new ArrayList<>();
```

Comment:
We don't need to define this list here since it has 2 usages, and they are in 2 different execution paths.